django/contrib/auth/base_user.py

```diff
@@ -77,6 +73,9 @@ def __init__(self, *args, **kwargs):
     def __str__(self):
         return self.get_username()
 
+    def clean(self):
+        setattr(self, self.USERNAME_FIELD, self.normalize_username(self.get_username()))
```

Comment:
Is there any case when someone use User class that inherits from `AbstractBaseUser` and don't define `USERNAME_FIELD`, since `USERNAME_FIELD` is only defined in `AbstractUser`